### PR TITLE
Simplified MvcRouteHandler code

### PR DESCRIPTION
**Pull request by ajaybhargavb**
#4291 

@rynowak

### CODE REVIEW

**Comment by rynowak** [comment]
No changes to tests? I know we have some tests for the logging/diagnostics part of this

**Comment by javiercn** [comment]
As far as I see, you've moved the preamble on the MvcRouteHandler into the action invoker, is that the case?

:shipit: when @rynowak is happy.

**Comment by ajaybhargavb** [comment]
Updated with test changes.

File: test/Microsoft.AspNetCore.Mvc.Core.Test/Internal/ControllerActionInvokerTest.cs [hunk 1]
```diff
@@ -2480,6 +2480,107 @@ public class ControllerActionInvokerTest
...
+                .Returns(displayName);
+            var actionDescriptor = mockActionDescriptor.Object;
+            actionDescriptor.MethodInfo = typeof(ControllerActionInvokerTest).GetMethod(
+                    nameof(ControllerActionInvokerTest.ActionMethod));
+            actionDescriptor.ControllerTypeInfo = typeof(ControllerActionInvokerTest).GetTypeInfo();
+            actionDescriptor.FilterDescriptors = new List<FilterDescriptor>();
+            actionDescriptor.Parameters = new List<ParameterDescriptor>();
+
+            var filter = Mock.Of<IFilterMetadata>();
+            var invoker = CreateInvoker(new[] { filter }, actionDescriptor, null, null, logger: logger);
```
Comment (dougbu) [inline test/Microsoft.AspNetCore.Mvc.Core.Test/Internal/ControllerActionInvokerTest.cs]: Shouldn't need the `null` arguments. If they are necessary, name them.

**Comment by javiercn** [comment]
:shipit:

**Pull request closed without merging**